pkg/container/container.go

```diff
@@ -0,0 +1,214 @@
+package container
+
+import (
+	"io"
+	"os"
+	"os/exec"
+	"os/user"
+	"strings"
+
+	"github.com/docker/docker/api/types/mount"
+
+	"hyscale/pkg/constants"
+	installer "hyscale/pkg/installer"
+	"hyscale/pkg/registry"
+	util "hyscale/pkg/util"
+)
+
+const (
+	//ImageCleanUpPolicy is used to clean the build images
+	ImageCleanUpPolicy = "IMAGE_CLEANUP_POLICY"
+
+	homeEnv            = "HYSCALECTL_HOME"
+	kubeConfEnv        = "HYSCALECTL_KUBECONF"
+	dockerConfEnv      = "HYSCALECTL_DOCKERCONF"
+	dockerConfigDirEnv = "DOCKER_CONFIG"
+	lbReadyTimeoutEnv  = "HYS_LB_READY_TIMEOUT"
+	//EqualsTo is used for constructing environment varibles
+	EqualsTo               = "="
+	containerFileSeparator = "/"
+	configDir              = "/.docker"
+	configJSONPath         = configDir + "/config.json"
+	kubeConfDir            = "/.kube"
+	kubeConfPath           = kubeConfDir + "/config"
+	unixSocket             = "/var/run/docker.sock"
+	dockerHost             = "DOCKER_HOST"
+	externalRegistryEnv    = "HYS_REGISTRY_CONFIG"
+)
+
+//TODO build image name with property
+var (
+	imageName      = "docker.io/hyscale/hyscale:0.9.6.0.39"
+	currentUser, _ = user.Current()
+	dockerConf     = currentUser.HomeDir + configJSONPath
+	dockerConfDir  = currentUser.HomeDir + configDir
+	kubeConf       = currentUser.HomeDir + kubeConfPath
+	pwd, _         = os.Getwd()
+)
+
+//HysContainer is container based installer for hyscale
+type HysContainer struct {
+}
+
+//Run method for Container based hyscale installer
+func (hyscontainer *HysContainer) Run(cliSpec *installer.DeploySpec) (resp *installer.DeployResponse, err error) {
+
+	labels := make(map[string]string)
+	labels["name"] = constants.Hyscale
+
+	args := []string{"run", "--rm"}
+
+	// Attach the cmd stdin to os.Stdin
+	args = append(args, "-i")
+
+	for k, v := range labels {
+		args = append(args, "--label")
+		args = append(args, BuildOption(k, v))
+	}
+	for k, v := range *getEnvs(constants.User) {
+		args = append(args, "-e")
+		args = append(args, BuildOption(k, v))
+	}
+
+	for _, mount := range *getVolumes() {
+		args = append(args, "-v")
+		args = append(args, BuildVolumes(&mount))
+	}
+	args = append(args, imageName)
+	args = append(args, os.Args[1:]...)
+
+	cmd := exec.Command("docker", args...)
+
+	// Setting the StdIn to standardinput in case of interative stream
+	if cliSpec.Interactive {
+		cmd.Stdin = os.Stdin
+	}
+
+	// Writing the docker config json to stdin of docker container
+	if len(cliSpec.Hspecs) > 0 {
+		authConfig, err := registry.GetAuthConfig(cliSpec.Hspecs)
+		if err != nil {
+			panic(err)
+		}
+		stdin, e := cmd.StdinPipe()
+		if e != nil {
+			panic(e)
+		}
+
+		go func() {
+			defer stdin.Close()
+			io.WriteString(stdin, *authConfig)
+		}()
+	}
+
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err = cmd.Run()
+	if err != nil {
+		if exit, ok := err.(*exec.ExitError); ok {
+			os.Exit(exit.ExitCode())
+		}
+	}
+	return resp, nil
```

Comment:
what does **resp** object contain in case if we want to run in background using "-d"?

Reply:
The response will still be independent of the command. It will deal with exit code only even when we run in background.